### ISSUE TITLE
Whether QWen in Cosyvoice2 trainable ?

### ISSUE THREAD

**Issue by n30nd** [FunAudioLLM/CosyVoice#1132]
I 'm finetuning Cosyvoice2 for Vietnamese dataset with script in examples/libritts/cosyvoice2 and meet overfit. 
With the same data, I fine tune flow and no meet overfit.
When I check conf/yaml file in cosyvoice and cosyvoice2, I saw that cosyvoice2 use llm encoder is qwen2 encoder and no have dropout in config, while llm of cosyvoice is detailed description. So. whether Qwen2 Encoder trainable? What I can do to solve overfit probleam?
Thank you for your time to read and answer my issue.

![Image](https://github.com/user-attachments/assets/8f38fb80-bab5-4544-88df-40453607f970)

**Comment by aluminumbox** [comment]
it is trainable

**Comment by n30nd** [comment]
> it is trainable

thank you for your answer. I see although train llm overfit, I give token out of val set (which get low accuracy when I train llm) to flow, then generated speech sound good. I see speech tokenizer is trained with Senvoice and language that not good with Vietnamese, is it deadlock ?

**Comment by aluminumbox** [comment]
> > it is trainable
> 
> thank you for your answer. I see although train llm overfit, I give token out of val set (which get low accuracy when I train llm) to flow, then generated speech sound good. I see speech tokenizer is trained with Senvoice and language that not good with Vietnamese, is it deadlock ?

no, you can try it on Vietnamese. no need to train speech tokenizer. It extracts more phoneme info, not just character info

**Comment by n30nd** [comment]
> > > it is trainable
> > 
> > 
> > thank you for your answer. I see although train llm overfit, I give token out of val set (which get low accuracy when I train llm) to flow, then generated speech sound good. I see speech tokenizer is trained with Senvoice and language that not good with Vietnamese, is it deadlock ?
> 
> no, you can try it on Vietnamese. no need to train speech tokenizer. It extracts more phoneme info, not just character info

Thank you. But I dont know why I got ovefit though I try use more data to finetune (1000h), or data only 1 speaker, all situation get overfit like image: val_accuracy increase to 31% then decrease . 

![Image](https://github.com/user-attachments/assets/68c5d4f9-e65e-4a74-b23b-e23e76c1d676)

I read paper cosyvoice2 in pattern "**Limitations**", it said that " **For languages with overlapping character sets, synthesis performance may degrade, presenting an open challenge for future research**", vietnamese and english has many same character but different phoneme, whether it is problem when I use your pretrained (in english) to finetune vietnamese data?

**Comment by aluminumbox** [comment]
> > > > it is trainable
> > > 
> > > 
> > > thank you for your answer. I see although train llm overfit, I give token out of val set (which get low accuracy when I train llm) to flow, then generated speech sound good. I see speech tokenizer is trained with Senvoice and language that not good with Vietnamese, is it deadlock ?
> > 
> > 
> > no, you can try it on Vietnamese. no need to train speech tokenizer. It extracts more phoneme info, not just character info
> 
> Thank you. But I dont know why I got ovefit though I try use more data to finetune (1000h), or data only 1 speaker, all situation get overfit like image: val_accuracy increase to 31% then decrease .
> 
> ![Image](https://github.com/user-attachments/assets/68c5d4f9-e65e-4a74-b23b-e23e76c1d676)
> 
> I read paper cosyvoice2 in pattern "**Limitations**", it said that " **For languages with overlapping character sets, synthesis performance may degrade, presenting an open challenge for future research**", vietnamese and english has many same character but different phoneme, whether it is problem when I use your pretrained (in english) to finetune vietnamese data?

llm dev acc 30%? are you training cosyvoice2 llm? you can add <|lid|> at sentence begin, llm will learn this pattern, with enough data, it can pronounce english with <|en|> and vietnamese with <|viet|> for example

**Comment by n30nd** [comment]
> > > > > it is trainable
> > > > 
> > > > 
> > > > thank you for your answer. I see although train llm overfit, I give token out of val set (which get low accuracy when I train llm) to flow, then generated speech sound good. I see speech tokenizer is trained with Senvoice and language that not good with Vietnamese, is it deadlock ?
> > > 
> > > 
> > > no, you can try it on Vietnamese. no need to train speech tokenizer. It extracts more phoneme info, not just character info
> > 
> > 
> > Thank you. But I dont know why I got ovefit though I try use more data to finetune (1000h), or data only 1 speaker, all situation get overfit like image: val_accuracy increase to 31% then decrease .
> > ![Image](https://github.com/user-attachments/assets/68c5d4f9-e65e-4a74-b23b-e23e76c1d676)
> > I read paper cosyvoice2 in pattern "**Limitations**", it said that " **For languages with overlapping character sets, synthesis performance may degrade, presenting an open challenge for future research**", vietnamese and english has many same character but different phoneme, whether it is problem when I use your pretrained (in english) to finetune vietnamese data?
> 
> llm dev acc 30%? are you training cosyvoice2 llm? you can add <|lid|> at sentence begin, llm will learn this pattern, with enough data, it can pronounce english with <|en|> and vietnamese with <|viet|> for example

thank you very much, let my try

**Comment by n30nd** [comment]
![Image](https://github.com/user-attachments/assets/1eedb274-c9b2-422e-a2e5-5f3c66d55e03)

![Image](https://github.com/user-attachments/assets/d07ec609-b45a-4263-8a56-26cb3881b78f)

I added <|vi|> and <|en|> to start of text but it seem not good, no overfit but underfit and I dont know why until 32%, every thing so strange, gradnorm increase very fast, val_acc decrease. Whether I miss any thing to work with Vietnamese ?

These are some information that I wonder
-----
I trained on my vietnamese single spk data with 3h, although get overfit on llm, i continue finetune flow, then check wer only 12%

----
In paper cosyvoice2, pattern 
**"2.7 Multi-Speaker Fine-tuning
Fine-tuning the pre-trained model on specific speakers (SFT) can further improve the generation
quality and speaker similarity. In this report, we introduce the multi-speaker fine-tuning (mSFT), in
which the pretrained model is fine-tuned on multiply speakers simultaneously rather than a single speaker. This approach ensures comprehensive prosody and pronunciation coverage across
multiple speakers and mitigates potential catastrophic forgetting from the pretrained models. To
avoid timbre confusion between various speakers, we prepend speaker-prompt tags, “Speaker
A<|endofprompt|>” to the input text for a specific speaker. If a training sample is not labeled
to a speaker, a special tag, “unknown<|endofprompt|>”, is utlized. The learning rate is set to 1e-5
during the whole multi-speaker fine-tuning process."**
Whether adding speaker-prompt tags is auto in source code? Is it help me to solve my problem with llm above, I still not have spk prompt tag in text input
